blocks/page_list/templates/itinerary/AddWalkToListDialog.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+export default class AddWalkToListDialog extends React.Component {
+  constructor(...args) {
+    super(...args);
+  }
+
+  propTypes() {
+    return {
+      lists: React.PropTypes.array,
+      add: React.PropTypes.func,
+      createList: React.PropTypes.func,
+      walkSelected: React.PropTypes.number,
+    }
+  }
+
+  render() {
+    const {lists, add, createList, activeWalk} = this.props;
+    //selectedWalk comes from where
+
+    const allLists = lists.map(list => {
+
+      const {id,title,walks} = list;
+
+      const walkFound = walks.find(walk => walk.id === activeWalk);
+
+      if(walkFound){
+        return ( //FIX: Issue with onClick, only available for last item in the list
+          <li onClick={(ev) => add(activeWalk,list,ev.target.value)} key={id}>
+            <span className="glyphicon glyphicon-ok">{title}</span>
+          </li>
+        )
+      } else {
+        return (
+          <li key={id}>
+            <span>{title}</span>
+          </li>
+        )
+      }
+    });
+
+    return(
+      <dialog id="addWalk">
+        <ul>
+          {allLists}
+        </ul>
+
+        <input ref="list" placeholder="Create a new List"></input>
+
+        <button onClick={(ev) => createList(walkSelected.id, this.refs.list.value)}>Create</button>
+        <button onClick={(ev) => document.getElementById('addWalk').close()}>Close</button>
```

Comment:
Don't rely on DOM selectors to bind events like this. Use refs instead: https://facebook.github.io/react/docs/more-about-refs.html

Also, closing a DOM el directly should be avoided, when it makes more sense to use the event to trigger a state change.